### ISSUE TITLE
App/window picker or drop down screen selection

### ISSUE THREAD

**Issue by LazyWings** [hyprwm/xdg-desktop-portal-hyprland#291]
I think the current implementation of xdg-desktop-portal has the following limitations:

-You can only screen capture entire screens rather than apps or windows (see KDE implementation)
-The selection is done with mouse, making it difficult to select a headless output which could have been a reasonable stopgap.

These two issues effectively prevent you from capturing specific windows/apps which has an impact on any screen recording application, e.g. OBS, Discord, Teams, or anything else you want to screen capture with.

Presumably you can't use the implementation in the KDE portal. I've tried manually adding in the 'org.freedesktop.impl.portal.AppChooser=kde' line to my config just incase but as expected, it has no effect since the portal implementation is more simplified. I think this is a limitation on wlr portals. Unfortunately I am FAR from experienced with xdg-desktop-portal.

Suggested change:
-Dedicated window/app chooser
-OR, drop down output selection instead of cursor picker.

Thanks

**Comment by fufexan** [comment]
xdph does support window/region sharing. Make sure you're not running xdp-wlroots.

**Comment by LazyWings** [comment]
Sigh... Ok user error on this one. I was using the wlr one, but it wasn't evidence. I uninstalled the wlr portals, reinstalled the hyprland-portals, redid my config file to only call hyprland portals. Works now. Feel a little stupid, but I didn't even know the screen picker existed. Sticking a picture of it on the readme would help. I only noticed it when I was digging around the files and so it in there.
